ledgercomm/interfaces/hid_device.py

```diff
@@ -53,13 +63,45 @@ def open(self) -> None:
         """
         if not self.__opened:
             if not self.path:
-                self.path = HID.enumerate_devices(self.vendor_id)[0]
+                self.path = HID._decide_device_path(self.vendor_id)
             self.device.open_path(self.path)
             self.device.set_nonblocking(True)
             self.__opened = True
 
     @staticmethod
-    def enumerate_devices(vendor_id: int = 0x2C97) -> List[bytes]:
+    def _decide_device_path(vendor_id: int = DEFAULT_VENDOR_ID) -> bytes:
+        if hid is None:
+            raise HIDAPINotInstalledError()
+
+        devices: List[Mapping[str, Any]] = list(hid.enumerate(vendor_id, 0))
+        LOG.debug("hid.enumerate(), devices: %s", devices)
+
+        devices = [device for device in devices if device.get("interface_number") == 0]
+        if len(devices) == 1:
+            # No ambiguity, pick the only device
+            return devices[0]["path"]
+
+        # On MacOS, "interface_number" is not a reliable filtering criteria, so we
+        # also filter by "usage_page".
+        devices = [device for device in devices if device.get(
+            "usage_page") == MACOS_USAGE_PAGE]
```

Comment:
Hello, this fails on Linux if two devices are plugged. I've fixed it with:
```suggestion
mac_filtered_devices = [device for device in devices if device.get(
             "usage_page") == MACOS_USAGE_PAGE]
        if len(mac_filtered_devices) != 0:
            # If we are on a MAC device, we keep the new filtered list
            devices = mac_filtered_devices
```